SSA/Experimental/ErasedContext.lean

```diff
@@ -101,13 +101,30 @@ theorem toSnoc_injective {Γ : Ctxt} {t t' : Ty} :
 
 abbrev Hom (Γ Γ' : Ctxt) := ⦃t : Ty⦄ → Γ.Var t → Γ'.Var t
 
+abbrev Hom.id {Γ : Ctxt} : Γ.Hom Γ :=
+  fun _ v => v
+
+/--
+  `map.with v₁ v₂` adjusts a single variable of a Context map, so that in the resulting map
+   * `v₁` now maps to `v₂`
+   * all other variables `v` still map to `map v` as in the original map
+-/
+def Hom.with {Γ₁ Γ₂ : Ctxt} (f : Γ₁.Hom Γ₂) {t : Ty} (v₁ : Γ₁.Var t) (v₂ : Γ₂.Var t) : Γ₁.Hom Γ₂ :=
+  fun t' w =>
+    if h : ∃ h : t = t', h ▸ w = v₁ then 
```

Comment:
why do you call these both `h`? it's a bit confusing, maybe you can call the two hypothesis differently for readibility?

Reply:
Good point, I just blindly copied this snippet, but it's indeed a bit confusing